### PR TITLE
Favorite page fetch liked posts FIX

### PR DESCRIPTION
**Pull request by fiskryeziu**
useEffect fix, added filter function to get the likedpost made by the currentUser. #788

### CODE REVIEW

File: src/components/Favorite.jsx [hunk 1]
```diff
@@ -13,46 +13,35 @@ function Favorite() {
...
-            variant: "error",
-          });
-        }
-      });
+    if (auth) {
+      const fetchLikedPosts = async () => {
+        setLoading(true);
+        const collections = collection(db, "posts");
+        console.log(collections);
+        const docSnap = await getDocs(collections);
```
Comment (narayan954) [inline src/components/Favorite.jsx:27]: isn't getting all the data from `posts` collection and then filtering very costly? This is also why we're using pagination as well?

File: src/components/Favorite.jsx [hunk 2]
```diff
@@ -13,46 +13,35 @@ function Favorite() {
...
-        setLoading(false);
-      } finally {
-        setLoading(false);
-      }
-    };
 
-    if (savedPostsArr.length === 0) {
-      setLoading(false);
+        const filteredPosts = likedPosts.filter((post) =>
+          post.likecount.includes(auth.currentUser.uid)
```
Comment (narayan954) [inline src/components/Favorite.jsx:35]: Also, favourite page is to get the posts we've saved, not liked.

Comment (narayan954) [reply src/components/Favorite.jsx:35]: there's an array perhaps savedpostsarray in the implementation that gets the list.

Comment (fiskryeziu) [reply src/components/Favorite.jsx:35]: thanks for the heads up, thought it was for the liked post

Comment (narayan954) [reply src/components/Favorite.jsx]: it's ok :)

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Comment by narayan954** [comment]
thanks for fixing this! :)

**Pull request merged**